vaadin-grid-flow-parent/vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -317,9 +314,6 @@ import { GridFlowSelectionColumn } from "./vaadin-grid-flow-selection-column.js"
               );
             }
           } else {
-            // workaround: sometimes grid-element gives page index that overflows
-            page = Math.min(page, Math.floor(grid.size / grid.pageSize));
```

Comment:
With the workaround, `rootPageCallbacks` can have a page with a callback of a different page (overflowing page), see line 338.